tests/test_clear_stream_source.py

```
@@ -0,0 +1,208 @@
+class FakeProcess:
+    """A child that streams the given stdout chunks, or answers communicate()."""
+
+    def __init__(self, chunks=(), output=b"", hang=False):
+        self.stdout = Stdout(chunks)
+        self.output = output
+        self.hang = hang
+        self.terminated = False
+        self.killed = False
+        self.input = None
+
+    def communicate(self, input=None, timeout=None):
```

Ruff (0.16.3)

[error] Function argument `input` is shadowing a Python builtin

(A002)